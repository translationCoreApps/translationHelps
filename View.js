
const api = window.ModuleApi;
const React = api.React;
const RB = api.ReactBootstrap;
const { Glyphicon } = RB;
const Markdown = require('react-remarkable');
const THelpsModal = require('./components/THelpsModal');
const style = require('./css/style');

class TranslationHelpsDisplay extends React.Component {
    render() {
        let { currentFile, modalFile } = this.props;
        if (currentFile) {
            return (
                <div style={style.translationHelpsContent}>
                  <style dangerouslySetInnerHTML={{
                    __html: [
                      '.remarkableStyling h1{',
                        'font-size: 19px;',
                        'font-weight: bold;',
                      '}',
                      '.remarkableStyling h2{',
                        'font-size: 14px;',
                        'font-weight: normal;',
                      '}'
                    ].join('\n')
                  }}>
                  </style>
                    <div onClick={this.props.showModal}>
                      <Glyphicon glyph={"new-window"}
                                 style={style.tHGlyphicon}/>
                    </div>
                    <THelpsModal show={this.props.modalVisibility}
                                 onHide={this.props.hideModal}>
                        <Markdown options={{html: true}} source={modalFile}/>
                    </THelpsModal>
                    <div className="remarkableStyling">
                        <Markdown options={{html: true}} source={currentFile} />
                    </div>
                </div>
            );
        }
        else {
            return (<div></div>);
        }
    }
}

module.exports = TranslationHelpsDisplay;
