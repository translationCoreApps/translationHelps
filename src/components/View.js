import React from 'react';
import { Glyphicon } from 'react-bootstrap';
import Markdown from 'react-remarkable';
import PropTypes from 'prop-types';
import THelpsModal from './THelpsModal';
import style from '../css/style';

class View extends React.Component {

  componentWillReceiveProps(nextProps) {
    if (this.props.currentFile !== nextProps.currentFile) {
      var page = document.getElementById("helpsbody");
      if (page) page.scrollTop = 0;
    }
  }

  render() {
    let { currentFile, modalFile } = this.props;
    if (currentFile) {
      return (
        <div style={{flex: 'auto', display: 'flex'}}>
          <style dangerouslySetInnerHTML={{
            __html: [
              '.remarkableStyling h1{',
              'font-size: 19px;',
              'font-weight: bold;',
              '}',
              '.remarkableStyling h2{',
              'font-size: 14px;',
              'font-weight: normal;',
              '}',
              '.remarkableStyling h3{',
              'font-size: 16px;',
              'font-weight: bold;',
              '}',
              '.remarkableStyling h4{',
              'font-size: 16px;',
              'font-weight: bold;',
              '}',
              '.remarkableStyling blockquote {',
              'font-size: small;',
              '}',
              '.remarkableStyling blockquote strong {',
              'text-decoration: underline;',
              'font-weight: normal;',
              '}'
            ].join('\n')
          }}>
          </style>
          <div style={style.helpsContent}>
            <div style={style.helpsHeader}>
              <Glyphicon glyph={"fullscreen"}
                         title="Click to show expanded helps"
                         onClick={this.props.showModal}
                         style={style.helpsIcon} />
            </div>
            <div id="helpsbody" className="remarkableStyling" style={style.helpsBody}>
              <Markdown options={{ html: true }} source={currentFile} />
            </div>
          </div>
          <THelpsModal show={this.props.modalVisibility}
            onHide={this.props.hideModal}>
            <Markdown options={{ html: true }} source={modalFile} />
          </THelpsModal>
        </div>
      );
    }
    else {
      return (<div></div>);
    }
  }
}

View.propTypes = {
  translate: PropTypes.func,
  currentFile: PropTypes.string,
  modalFile: PropTypes.string,
  modalVisibility: PropTypes.bool.isRequired,
  showModal: PropTypes.func.isRequired,
  hideModal: PropTypes.func.isRequired
};

export default View;
