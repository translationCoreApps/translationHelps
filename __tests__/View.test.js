/* eslint-env jest */
import React from 'react';
import View from '../src/components/View';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const props = {
  translate:k=>k,
  currentFile: '# current file #\n\nHere is the current file content',
  modalFile: '# modal file #\n\nHere is the modal file content',
  modalVisibility: true,
  showModal: jest.fn(),
  hideModal: jest.fn()
};

describe('View component Tests', () => {
  it('Check View component', () => {
    const wrapper = shallow(<View {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Test View.componentWillReceiveProps', () => {
    const wrapper = shallow(<View {...props} />);
    document.getElementById('test');
    expect(wrapper.instance().props.currentFile).toEqual(props.currentFile);
    const newCurrentFile = '# new current file #\n\nnew current file contents';
    const nextProps = {
      ...props,
      currentFile: newCurrentFile
    };
    wrapper.setProps(nextProps);
    expect(wrapper.instance().props.currentFile).toEqual(newCurrentFile);
  });
});
