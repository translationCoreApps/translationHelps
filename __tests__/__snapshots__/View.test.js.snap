// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View component Tests Check View component 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flex": "auto",
    }
  }
>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".remarkableStyling h1{
font-size: 19px;
font-weight: bold;
}
.remarkableStyling h2{
font-size: 14px;
font-weight: normal;
}
.remarkableStyling h3{
font-size: 16px;
font-weight: bold;
}
.remarkableStyling h4{
font-size: 16px;
font-weight: bold;
}
.remarkableStyling blockquote {
font-size: small;
}
.remarkableStyling blockquote strong {
text-decoration: underline;
font-weight: normal;
}",
      }
    }
  />
  <div
    style={
      Object {
        "backgroundColor": "var(--background-color-light)",
        "display": "flex",
        "flexDirection": "column",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "borderBottom": "1px solid var(--border-color)",
          "display": "flex",
          "flex": "0 0 50px",
          "justifyContent": "flex-end",
          "padding": "0 15px",
        }
      }
    >
      <Glyphicon
        bsClass="glyphicon"
        glyph="fullscreen"
        onClick={[Function]}
        style={
          Object {
            "color": "var(--text-color)",
            "cursor": "pointer",
            "fontSize": "20px",
          }
        }
        title="Click to show expanded helps"
      />
    </div>
    <div
      className="remarkableStyling"
      id="helpsbody"
      style={
        Object {
          "overflowY": "auto",
          "padding": "0 15px 30px",
        }
      }
    >
      <Remarkable
        container="div"
        options={
          Object {
            "html": true,
          }
        }
        source="# current file #

Here is the current file content"
      />
    </div>
  </div>
  <THelpsModal
    onHide={[Function]}
    show={true}
  >
    <Remarkable
      container="div"
      options={
        Object {
          "html": true,
        }
      }
      source="# modal file #

Here is the modal file content"
    />
  </THelpsModal>
</div>
`;
